app/services/combined_audio_derivative_creator.rb

```diff
@@ -0,0 +1,163 @@
+require 'tempfile'
+#
+# Generate the combined audio derivatives for a
+# given oral history work.
+#
+# some_work = Work.find_by_friendlier_id(friendlier_id)
+# CombinedAudioDerivativeCreator.new(some_work).generate
+#
+#
+# Sample output:
+#     {
+#      :combined_audio_mp3_data=>
+#          "/var/folders/n7/t4zt2w751bj_6pbnf95sgpnrmz8lz8/T/output20200228-61637-ec6kjz.mp3",
+#      :combined_audio_webm_data=>
+#          "/var/folders/n7/t4zt2w751bj_6pbnf95sgpnrmz8lz8/T/output20200228-61637-1btjscz.webm",
+#      :fingerprint=>"4998d366f5edb6222db1181c7b153e21",
+#      :component_metadata=>
+#        {:durations=>
+#          [
+#            "00:30:30.29",
+#            "00:29:13.32",
+#            "00:33:03.26",
+#            "00:24:28.94"
```

Comment:
So, these free floating times makes me a bit worried that we'd get confused about which one goes with which file -- but I guess the fingerprint (unless it's buggy) ensures that these go with a certain list of files in a certain order. 

Still, can we make this something that is actually pairs of ids representing files to times, instead of just times? That will make it easier to access in the access patterns we're going to need too -- we're going to have a file, and want to look up it's timecode in order to add a link on that file to "jump to" a certain timecode. 

That can probably be returned as a hash -- hashes are ordered in ruby, ordered by insertion order. Or it could be an array of two-element arrays `[[id, time], [id, time]]` (which makes it easier to keep in order, and can be easily turned into a hash). 

The ids can be actual UUID pks, or friendlier_ids. Friendlier_ids are a lot easier to deal with, but also are hypothetically mutable, at which point they'd invalidate the data. I wonder if it actually makes sense just to use the UUIDs, they shoudln't change and are more universally available internal to the app. 

**Wait though**, also, "durations" aren't really what we want -- we want START TIMES. So actually calculate them for us, they are what we want. (ie, start time of 3rd one is duration1+duration2).  And then call it :start_times instead of :durations.

Reply:
OK  -- returning :start_times -- just a running total of durations, minus the last duration which I suppose we don't care about. Keys are UUIDs (not friendlier IDs). Good call.